pipenv/project.py

```diff
@@ -574,32 +578,55 @@ def clear_pipfile_cache(self):
         _pipfile_cache.clear()
 
     def _parse_pipfile(self, contents):
+        toml_encoder = toml.TomlPreserveInlineDictEncoder()
+        toml_encoder.get_empty_table()
         # If any outline tables are present...
         if ("[packages." in contents) or ("[dev-packages." in contents):
-            data = toml.loads(contents)
+            data = tomlkit.parse(contents)
             # Convert all outline tables to inline tables.
             for section in ("packages", "dev-packages"):
                 for package in data.get(section, {}):
                     # Convert things to inline tables — fancy :)
                     if hasattr(data[section][package], "keys"):
+                        table = tomlkit.inline_table()
                         _data = data[section][package]
-                        data[section][package] = toml.TomlDecoder().get_empty_inline_table()
-                        data[section][package].update(_data)
-            toml_encoder = toml.TomlEncoder(preserve=True)
+                        table.update(_data)
+                        data[section][package] = table
             # We lose comments here, but it's for the best.)
             try:
-                return contoml.loads(toml.dumps(data, encoder=toml_encoder))
+                return data
 
             except RuntimeError:
-                return toml.loads(toml.dumps(data, encoder=toml_encoder))
+                return toml.loads(tomlkit.dumps(data, encoder=toml_encoder))
```

Comment:
Only `return data` here, it is safe to remove the `try except` block.

Reply:
awesome, thanks for the review => I also patched tomlkit which I am a lot more unsure about